File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        dummy = {}
        for num in set(arr):
            dummy[num] = arr.count(num)

        st = set()

        for val in dummy.values():
            if val in st:
                return False
            else:
                st.add(val)

        return True
        

File: Easy/1319-unique-number-of-occurrences/unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer,Integer> map = new HashMap<>();
        for(int i:arr){
            if(map.containsKey(i)){
                int temp = map.get(i);
                map.put(i,++temp);
            }
            else{
                map.put(i,1);
            }
        }
        Set<Integer> set = new HashSet<>();
        for(int i:map.values()){
            if(!set.add(i)){
                return false;
            }
        }
    return true;
    }
}